> Would you be interested in adding some commits to ensure that the CI tests pass? If you'd prefer not to write that code, that's perfectly fine—I can go ahead and merge as is.

These are lint errors:
```
executor/engine/job/utils.py:56: error: Name "_thread_locals" is not defined  [name-defined]
executor/engine/job/utils.py:5[8](https://github.com/Nanguage/executor-engine/actions/runs/11024907333/job/30618876230?pr=6#step:6:9): error: No overload variant of "next" matches argument type "Future[Any]"  [call-overload]
executor/engine/job/utils.py:58: note: Possible overload variants:
executor/engine/job/utils.py:58: note:     def [_T] next(SupportsNext[_T], /) -> _T
executor/engine/job/utils.py:58: note:     def [_T, _VT] next(SupportsNext[_T], _VT, /) -> Union[_T, _VT]
executor/engine/job/utils.py:64: error: Name "_thread_locals" is not defined  [name-defined]
executor/engine/job/utils.py:66: error: "Future[Any]" has no attribute "__anext__"  [attr-defined]
Found 4 errors in 1 file (checked 22 source files)
```
1. the `_thread_locals` variable is dynamically defined because I don't want to mix it in the main thread.
2. the `Future.__anext__` method exists only for future returned by DaskJob, I'm not sure howto fix this lint error.

--- a/executor/engine/job/utils.py
+++ b/executor/engine/job/utils.py
@@ -10,7 +10,6 @@ from ..utils import CheckAttrRange, ExecutorError
 if T.TYPE_CHECKING:
     from .base import Job
 
-
 JobStatusType = T.Literal['pending', 'running', 'failed', 'done', 'cancelled']
 valid_job_statuses: T.List[JobStatusType] = [
     'pending', 'running', 'failed', 'done', 'cancelled']
@@ -40,7 +39,7 @@ class InvalidStateError(ExecutorError):
 
 
 _T = T.TypeVar("_T")
-
+_thread_locals = threading.local()
 
 def _gen_initializer(gen_func, args=tuple(), kwargs={}):  # pragma: no cover
     global _thread_locals
@@ -50,7 +49,7 @@ def _gen_initializer(gen_func, args=tuple(), kwargs={}):  # pragma: no cover
     _thread_locals._generator = gen_func(*args, **kwargs)
 
 
-def _gen_next(fut: T.Optional[Future] = None):  # pragma: no cover
+def _gen_next(fut = None):  # pragma: no cover
     global _thread_locals
     if fut is None:
         return next(_thread_locals._generator)
@@ -58,7 +57,7 @@ def _gen_next(fut: T.Optional[Future] = None):  # pragma: no cover
         return next(fut)
 
 
-def _gen_anext(fut: T.Optional[Future] = None):  # pragma: no cover
+def _gen_anext(fut = None):  # pragma: no cover
     global _thread_locals
     if fut is None:
         return asyncio.run(_thread_locals._generator.__anext__())

--- a/executor/engine/job/utils.py
+++ b/executor/engine/job/utils.py
@@ -41,6 +41,7 @@ class InvalidStateError(ExecutorError):
 _T = T.TypeVar("_T")
 _thread_locals = threading.local()
 
+
 def _gen_initializer(gen_func, args=tuple(), kwargs={}):  # pragma: no cover
     global _thread_locals
     if "_thread_locals" not in globals():
@@ -49,7 +50,7 @@ def _gen_initializer(gen_func, args=tuple(), kwargs={}):  # pragma: no cover
     _thread_locals._generator = gen_func(*args, **kwargs)
 
 
-def _gen_next(fut = None):  # pragma: no cover
+def _gen_next(fut=None):  # pragma: no cover
     global _thread_locals
     if fut is None:
         return next(_thread_locals._generator)
@@ -57,7 +58,7 @@ def _gen_next(fut = None):  # pragma: no cover
         return next(fut)
 
 
-def _gen_anext(fut = None):  # pragma: no cover
+def _gen_anext(fut=None):  # pragma: no cover
     global _thread_locals
     if fut is None:
         return asyncio.run(_thread_locals._generator.__anext__())
@@ -69,6 +70,7 @@ class GeneratorWrapper(T.Generic[_T]):
     """
     wrap a generator in executor pool
     """
+
     def __init__(self, job: "Job", fut: T.Optional[Future] = None):
         self._job = job
         self._fut = fut

